drivers/sensor/f75303/f75303.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT fintek_f75303
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/sensor/f75303.h>
+#include "f75303.h"
+
+LOG_MODULE_REGISTER(F75303, CONFIG_SENSOR_LOG_LEVEL);
+
+static int f75303_fetch(const struct i2c_dt_spec *i2c,
+			uint8_t off_h, uint8_t off_l, uint16_t *sample)
+{
+	uint8_t val_h;
+	uint8_t val_l;
+	int res;
+
+	res = i2c_reg_read_byte_dt(i2c, off_h, &val_h);
+	if (res) {
+		return res;
+	}
+
+	res = i2c_reg_read_byte_dt(i2c, off_l, &val_l);
+	if (res) {
+		return res;
+	}
+
+	*sample = val_h << 3 | val_l >> 5;
+
+	return 0;
+}
+
+static int f75303_fetch_local(const struct device *dev)
+{
+	struct f75303_data *data = dev->data;
+	const struct f75303_config *config = dev->config;
+
+	return f75303_fetch(&config->i2c,
+			    F75303_LOCAL_TEMP_H,
+			    F75303_LOCAL_TEMP_L,
+			    &data->sample_local);
+}
+
+static int f75303_fetch_remote1(const struct device *dev)
+{
+	struct f75303_data *data = dev->data;
+	const struct f75303_config *config = dev->config;
+
+	return f75303_fetch(&config->i2c,
+			    F75303_REMOTE1_TEMP_H,
+			    F75303_REMOTE1_TEMP_L,
+			    &data->sample_remote1);
+}
+
+static int f75303_fetch_remote2(const struct device *dev)
+{
+	struct f75303_data *data = dev->data;
+	const struct f75303_config *config = dev->config;
+
+	return f75303_fetch(&config->i2c,
+			    F75303_REMOTE2_TEMP_H,
+			    F75303_REMOTE2_TEMP_L,
+			    &data->sample_remote2);
+}
+
+static int f75303_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	int res;
+
+	switch ((uint32_t)chan) {
+	case SENSOR_CHAN_ALL:
+		res = f75303_fetch_local(dev);
+		if (res) {
+			break;
+		}
+		res = f75303_fetch_remote1(dev);
+		if (res) {
+			break;
+		}
+		res = f75303_fetch_remote2(dev);
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		return f75303_fetch_local(dev);
+	case SENSOR_CHAN_F75303_REMOTE1:
+		return f75303_fetch_remote1(dev);
+	case SENSOR_CHAN_F75303_REMOTE2:
+		return f75303_fetch_remote2(dev);
+	default:
+		return -ENOTSUP;
+	}
+
+	return res;
+}
+
+static int f75303_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct f75303_data *data = dev->data;
+	uint16_t sample;
+
+	switch ((uint32_t)chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		sample = data->sample_local;
+		break;
+	case SENSOR_CHAN_F75303_REMOTE1:
+		sample = data->sample_remote1;
+		break;
+	case SENSOR_CHAN_F75303_REMOTE2:
+		sample = data->sample_remote2;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	val->val1 = sample >> 3;
+	val->val2 = (sample & 7) * 125000;
```

Comment:
Please add a comment documenting this conversion.

Reply:
Would also be nice to add some constants for the numbers, e.g.

```
#define F75303_TEMP_SAMPLE_INT_SHIFT (3)
#define F75303_TEMP_SAMPLE_FRAC_MASK GENMASK(2,0)
#define F75303_TEMP_SAMPLE_MICROCELSIUS_PER_BIT (125000)
```

These don't need to go in the header neither tests nor applications need these.